org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -84,8 +86,9 @@ public void initializeProjects(final Collection<IPath> rootPaths, IProgressMonit
 			public void run(IProgressMonitor monitor) throws CoreException {
 				SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
 				deleteInvalidProjects(rootPaths, subMonitor.split(10));
+				cleanGradleModel(subMonitor.split(10));
```

Comment:
I'd rather keep ProjectsManager decoupled from buildship(and m2e). Can you maybe delegate to the BuildSupport classes (some new workspaceInitialization() method)?